File: src/main/java/com/svantle/score/commands/WorkbenchCommand.java
package com.svantle.score.commands;

import com.svantle.score.utils.MessagesHandler;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class WorkbenchCommand implements CommandExecutor {

    private final MessagesHandler messagesHandler;

    public WorkbenchCommand(MessagesHandler messagesHandler) {
        this.messagesHandler = messagesHandler;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        if (!(sender instanceof Player player)) {
            sender.sendMessage(messagesHandler.prefixed(messagesHandler.get("only_players")));
            return true;
        }

        if (!player.hasPermission("score.workbench")) {
            player.sendMessage(messagesHandler.prefixed(messagesHandler.get("no_permission")));
            return true;
        }

        player.openWorkbench(null, true);
        player.sendMessage(messagesHandler.prefixed(messagesHandler.get("workbench_opened")));

        return true;
    }
}

File: src/main/kotlin/com/svantle/score/commands/WorkbenchCommand.kt
package com.svantle.score.commands

import com.svantle.score.SCore
import com.svantle.score.constants.*
import com.svantle.score.utils.MessageHelper
import com.svantle.score.utils.sendTo
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class WorkbenchCommand(core: SCore) : CommandExecutor {

	init {
		core.getCommand("workbench")!!.setExecutor(this)
	}

	override fun onCommand(sender: CommandSender, cmd: Command, label: String, args: Array<String>?): Boolean {
		if (sender !is Player) {
			MessageHelper.phrase(MSG_ONLY_PLAYERS).sendTo(sender)
		} else {
			if (sender.hasPermission(WORKBENCH_COMMAND)) {
				sender.openWorkbench(null, true)
				MessageHelper.phrase(MSG_WORKBENCH_OPENED).sendTo(sender)
			} else {
				MessageHelper.phrase(MSG_NO_PERMISSION).sendTo(sender)
			}
		}

		return true
	}
}
